tsdb/tsm1/predicate.go

```diff
@@ -386,9 +387,12 @@ func newPredicateCache(state *predicateState) predicateCache {
 }
 
 // Clone returns a deep copy of p.
-func (p *predicateCache) Clone() *predicateCache {
+func (p *predicateCache) Clone(state *predicateState) *predicateCache {
```

Comment:
I think it's probably worth documenting why `state` needs to be pushed down into this.